modules/roles_profiles/manifests/profiles/notarization_users.pp

```diff
@@ -0,0 +1,54 @@
+# This Source Code Form is subject to the terms of the Mozilla Public
+# License, v. 2.0. If a copy of the MPL was not distributed with this
+# file, You can obtain one at http://mozilla.org/MPL/2.0/.
+
+class roles_profiles::profiles::notarization_users {
+    case $::operatingsystem {
+        'Darwin': {
+            include sudo
+            # Make sure the users profile is required
+            # That is where the user virtual resources are generated
+            require roles_profiles::profiles::users
+            # Lookup the group array and realize their user resource
+            $users = $::hostname ? {
+                /^(tb-)?mac-v3-signing\d+/ =>
+                    lookup('user_groups.notarization', Array, undef, []),
+                /^dep-mac-v3-signing\d+/ =>
+                    lookup('user_groups.dep_notarization', Array, undef, undef),
```

Comment:
Is this a preferred method over having a settings.pp with the various settings specified per group/trustlevel, then specifying the group or trustlevel via hostname regex?

Reply:
It's the method used for our own accounts, which are arguably the same importance.

Reply:
> > > > * a secrets solution :)
> 
> > > Out of scope for this PR, I think
> > 
> > 
> > We need it before we can use this PR to image our machines, no? And since Relops uses ronin-puppet to half-puppetize our machines after imaging, if we land this and don't have the secrets to populate, we may cause problems.
> 
> There's currently a set of hiera secrets that are copied out of vault and onto the node itself for deployment. Is that enough, or does it need more? Because I don't think I can do more without relops deploying something.

If we have a way of updating that set of hiera secrets that gets deployed onto the node, I consider that a secrets solution.

> > > I'm not sure how puppet could clone a private repo
> > 
> > 
> > a) land ssh key somewhere, b) use ssh key to `git clone` ?
> 
> Urgh. Yes, but also urgh. This is getting very hacky.

I think my preferred long term solution is a non-public s3-or-similar url where we publish the widevine wheel. As long as it's internal-only, we can download it and install it.

> > We will be adjusting these settings in the future. As it stands, we'll need to find all files that are applicable and adjust them, as opposed to adjusting the settings.pp file.
> 
> I'm not sure how the different users to be deployed would work, in that scenario, and the certificates (given we can't map hostname to value directly due to yaml restrictions)

[Here](https://github.com/mozilla-releng/build-puppet/blob/master/modules/signing_scriptworker/manifests/settings.pp#L31-L45) we have a nickname, `dep`, that specifies configuration. The users to be deployed could be a list in that dict, no? Then the hostname regex would only need to specify that `dep-mac...` is `dep` configuration, and we would have a single location to update our configs.

Reply:
That nickname is mostly for the templates at the moment, I think it would be confusing to overload that and then not use a lot of the variables in the templates. 

`dep` doesn't just get a list of users, but each user has its own suffix and directory. I can see what a config area looks like, but I suspect it will be harder to read.